opal/mca/btl/usnic/btl_usnic_recv.c

```diff
@@ -340,8 +340,9 @@ void opal_btl_usnic_recv_call(opal_btl_usnic_module_t *module,
         opal_output(0, "    Received ACK for sequence number %" UDSEQ " from %s to %s\n",
                     bseg->us_btl_header->ack_seq, remote_ip, local_ip);
 #endif
+        OPAL_THREAD_LOCK(&btl_usnic_send_lock);
         opal_btl_usnic_handle_ack(endpoint, ack_seq);
-
+        OPAL_THREAD_UNLOCK(&btl_usnic_send_lock);
```

Comment:
It's a bit odd that the lock is named "send_lock" when you also use it in the receive.  I'm guessing this is left over from when you had multiple locks.  Can you rename the lock to be something more generic -- perhaps even `btl_usnic_lock`?